dGame/dComponents/PropertyEntranceComponent.cpp

```diff
@@ -74,261 +67,222 @@ void PropertyEntranceComponent::OnEnterProperty(Entity* entity, uint32_t index,
 	}
 
-	delete playerPropertyLookupResults;
-	playerPropertyLookupResults = nullptr;
-
-	delete playerPropertyLookup;
-	playerPropertyLookup = nullptr;
-
 	entries.push_back(playerEntry);
 
-	const auto query = BuildQuery(entity, sortMethod, character);
-
-	auto propertyLookup = Database::CreatePreppedStmt(query);
-
-	const auto searchString = "%" + filterText + "%";
-	propertyLookup->setUInt(1, this->m_MapID);
-	propertyLookup->setString(2, searchString.c_str());
-	propertyLookup->setString(3, searchString.c_str());
-	propertyLookup->setString(4, searchString.c_str());
-	propertyLookup->setInt(5, sortMethod == SORT_TYPE_FEATURED || sortMethod == SORT_TYPE_FRIENDS ? (uint32_t)PropertyPrivacyOption::Friends : (uint32_t)PropertyPrivacyOption::Public);
-	propertyLookup->setInt(6, numResults);
-	propertyLookup->setInt(7, startIndex);
+	auto propertyIds = this->GetPropertyIDsBasedOnParams(filterText, sortMethod, entity);
 
-	auto propertyEntry = propertyLookup->executeQuery();
+	std::vector<uint32_t> propertyIdsSlice(propertyIds.begin() + startIndex, propertyIds.begin() + std::min(startIndex + numResults, (int32_t)propertyIds.size()));
 
-	while (propertyEntry->next()) {
-		const auto propertyId = propertyEntry->getUInt64(1);
-		const auto owner = propertyEntry->getInt(2);
-		const auto cloneId = propertyEntry->getUInt64(4);
-		const auto propertyNameFromDb = std::string(propertyEntry->getString(5).c_str());
-		const auto propertyDescriptionFromDb = std::string(propertyEntry->getString(6).c_str());
-		const auto privacyOption = propertyEntry->getInt(9);
-		const auto modApproved = propertyEntry->getBoolean(10);
-		const auto dateLastUpdated = propertyEntry->getInt(11);
-		const float reputation = propertyEntry->getInt(14);
-		const auto performanceCost = (float)propertyEntry->getDouble(16);
+	std::vector<uint32_t> propertyCloneIdSlice{};
 
-		PropertySelectQueryProperty entry{};
+	for (const auto& id : propertyIdsSlice) {
+		auto prop = this->GetPropertyData(id);
+		if (prop.CloneID != 0) {
+			prop.PersonalData = this->GetPropertyPersonalData(prop, entity, true);
+			entries.push_back(prop);
+		}
+		propertyCloneIdSlice.push_back(prop.CloneID);
+	}
 
-		std::string ownerName = "";
-		bool isOwned = true;
-		auto nameLookup = Database::CreatePreppedStmt("SELECT name FROM charinfo WHERE prop_clone_id = ?;");
+	this->m_UserRequestedCloneMap[entity->GetObjectID()] = propertyCloneIdSlice;
 
-		nameLookup->setUInt64(1, cloneId);
+	GameMessages::SendPropertySelectQuery(m_Parent->GetObjectID(), startIndex, propertyIds.size() > propertyIdsSlice.size() + startIndex, character->GetPropertyCloneID(), false, true, entries, sysAddr);
+}
 
-		auto nameResult = nameLookup->executeQuery();
 
-		if (!nameResult->next()) {
-			delete nameLookup;
-			nameLookup = nullptr;
+PropertyData PropertyEntranceComponent::GetPropertyData(uint32_t propertyID) {
+	if (m_PropertyDataCache.find(propertyID) != m_PropertyDataCache.end()) {
+		return m_PropertyDataCache[propertyID];
```

Comment:
if a user updates their property name, description or has their player name updated while the data here is cached, when is this updated?